lib/internal/url.js

```diff
@@ -105,7 +105,11 @@ function parse(input, base) {
       this[context].query = query;
       this[context].fragment = fragment;
       this[context].host = host;
-      this[searchParams] = new URLSearchParams(query);
+      if (this[searchParams]) {  // invoked from href setter
+        initSearchParams(this[searchParams], query);
```

Comment:
I am not very sure about the possibility of the leading `?`. If we do need to make it a precondition, then we probably need to add a comment in `node_url.cc`?